tests/test_export.py

```diff
@@ -392,9 +395,25 @@ def test_export_data(mocker):
     assert pyhf.writexml._ROOT_DATA_FILE.__setitem__.called
 
 
+def test_export_root_histogram(mocker, tmp_path):
+    """
+    Test that pyhf.writexml._export_root_histogram writes out a histogram
+    in the manner that uproot is expecting
+    """
+    mocker.patch("pyhf.writexml._ROOT_DATA_FILE", {})
+    pyhf.writexml._export_root_histogram("example", [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+
+    with uproot.recreate(tmp_path.joinpath("test_export_root_histogram.root")) as file:
+        file["hist"] = pyhf.writexml._ROOT_DATA_FILE["example"]
+        values, edges = file["hist"].to_numpy()
```

Comment:
This literally writes and reads the data back. If you want to keep an Uproot Model of the data for later processing, you could use `uproot.to_writable` to turn it into a Model, which is implicitly the first step in writing anyway.

Reply:
This was more just meant as a very simple sanity check that `pyhf.writexml._export_root_histogram` is writing what we expect it to into the ROOT file. I see though that

```python
>>> import uproot
>>> import numpy as np
>>> data = np.arange(10)
>>> edges = np.arange(len(data) + 1)
>>> uproot.to_writable((data, edges))
<TH1D (version 3) at 0x7f421bf86df0>
```

verifies the structure. So in your mind should I add the suggestion that I've placed in `pyhf.writexml._export_root_histogram`?

Reply:
If it's a sanity check, then writing and reading back is a good idea. It would be even more of a sanity check if you did so with different file objects:

```python
    with uproot.recreate(tmp_path.joinpath("test_export_root_histogram.root")) as file:
        file["hist"] = pyhf.writexml._ROOT_DATA_FILE["example"]

    with uproot.open(tmp_path.joinpath("test_export_root_histogram.root")) as file:
        values, edges = file["hist"].to_numpy()
```

That would test more—the whole file structure, and the certainty that these really are bytes on disk. Using the same `file` does invoke the system operation to read from the file it's just written to, but with the same file handle in `"r+b"` mode and without going back to the beginning of the file and seeking through all the pointers to get to the object again. `to_writable` does still less, just turning the 2-tuple of arrays Python object into an Uproot Model Python object, which is what you'd want to do if performance were your goal, rather than sanity-checking.

So it's up to you, depending on what your goal is here.

Reply:
Good point. I've added this in the latest commit.